src/server/acl/helpers.cc

```diff
@@ -79,6 +79,47 @@ std::string PrettyPrintSha(std::string_view pass, bool all) {
   return absl::BytesToHexString(pass.substr(0, 15)).substr(0, 15);
 };
 
+std::optional<ParseKeyResult> MaybeParseAclKey(std::string_view command) {
+  if (absl::EqualsIgnoreCase(command, "ALLKEYS") || command == "~*") {
+    return ParseKeyResult{"", {}, true};
+  }
+
+  if (absl::EqualsIgnoreCase(command, "RESETKEYS")) {
+    return ParseKeyResult{"", {}, false, true};
+  }
+
+  auto op = KeyOp::READ_WRITE;
+
+  if (absl::StartsWith(command, "%")) {
+    if (command.size() < 3) {
+      return {};
+    }
+
+    if (!(command.substr(1, 2) == "RW")) {
+      if (command[1] == 'R') {
+        op = KeyOp::READ;
+      } else if (command[1] == 'W') {
+        op = KeyOp::WRITE;
+      } else {
+        return {};
+      }
+      command = command.substr(2);
+    } else {
+      command = command.substr(3);
+    }
+  }
```

Comment:
Maybe checking 3 times startswith would  even be easier 😅 , because I had to think at the `if (! (substr == "RW"))` part 🤣 (`!=`)

Reply:
so much better now!